receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +90,489 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+		datapointAttributes := pcommon.NewMap()
+		for _, attribute := range metricAttributes {
+			attributeCfg := cfg.Attributes[attribute.Name]
+			attributeKey := attribute.Name
+			if attributeCfg.Value != "" {
+				attributeKey = attributeCfg.Value
+			}
+			attributeValue := attribute.Value
+			if attributeCfg.IndexedValuePrefix != "" {
+				attributeValue = attributeCfg.IndexedValuePrefix + indexString
+			} else if attributeCfg.OID != "" {
+				attrKey := indexedAttributeKey{
+					parentOID: attributeCfg.OID,
+					oidIndex:  indexString,
+				}
+				attributeValue = indexedAttributeValues[attrKey]
+				if attributeValue == "" {
+					return fmt.Errorf(errMsgOIDAttributeEmptyValue, metricName)
+				}
+			}
+			datapointAttributes.PutString(attributeKey, attributeValue)
+		}
+
+		resourceAttributes := map[string]string{}
+		resourceAttributeNames := []string{}
+		// Create a map of key/values for all related resource attributes. Keys will come directly from the
+		// metric config's resource attribute values. Values will come from the related attribute config's
+		// prefix value plus the index OR the previously collected resource attribute indexed data.
+		for _, attributeName := range metricResourceAttributes {
+			resourceAttributeNames = append(resourceAttributeNames, attributeName)
+			resourceAttributeCfg := cfg.ResourceAttributes[attributeName]
+
+			var attributeValue string
+			if resourceAttributeCfg.IndexedValuePrefix != "" {
+				attributeValue = resourceAttributeCfg.IndexedValuePrefix + indexString
+			} else if resourceAttributeCfg.OID != "" {
+				attrKey := indexedAttributeKey{
+					parentOID: resourceAttributeCfg.OID,
+					oidIndex:  indexString,
+				}
+				attributeValue = indexedResourceAttributeValues[attrKey]
+				if attributeValue == "" {
+					return fmt.Errorf(errMsgOIDResourceAttributeEmptyValue, metricName)
+				}
+			}
+			resourceAttributes[attributeName] = attributeValue
+		}
+
+		// Create a resource key using all of the relevant resource attribute names
+		sort.Strings(resourceAttributeNames)
+		resourceKey := ""
+		if len(resourceAttributeNames) > 0 {
+			resourceKey = strings.Join(resourceAttributeNames, "...") + indexString
+		}
+		// If this resource exists, then grab the relevant metric slice to put new metric data on
+		// If it doesn't, then create a new ResourceMetrics with the resource metric map that we
+		// just created and use the newly created metric slice for new metric data
+		var metricSlice pmetric.MetricSlice
+		if resourceMetricsMap[resourceKey] != nil {
+			metricSlice = resourceMetricsMap[resourceKey].ScopeMetrics().At(0).Metrics()
+		} else {
+			resourceMetrics := pmetric.NewResourceMetrics()
+			for key, value := range resourceAttributes {
+				resourceMetrics.Resource().Attributes().PutString(key, value)
+			}
+			scopeMetrics := resourceMetrics.ScopeMetrics().AppendEmpty()
+			scopeMetrics.Scope().SetName("otelcol/snmpreceiver")
```

Comment:
I couldn't find where this is done for the generic resource.

Reply:
@StefanKurek for the generic one this is done right up front in the `scrape()` function